File: frontend/src/client/service/user.dto.ts
export type UserDto = {
  id: string
  email: string
  isActive: boolean
  isOrganizationAdmin: boolean
  fullName: string
  createdAt: string
  updatedAt: string
}

export type UserUpdateMeDto = {
  fullName: string
}

export type UserUpdatePasswordMeDto = {
  currentPassword: string
  newPassword: string
}

File: backend/src/main/java/com/gistacsl/mosaic/user/dto/UserDto.java
package com.gistacsl.mosaic.user.dto;

import java.time.OffsetDateTime;
import java.util.UUID;

public class UserDto {

  public record Res(
      UUID id,
      String email,
      Boolean isActive,
      Boolean isOrganizationAdmin,
      String fullName,
      String organizationName,
      OffsetDateTime createdAt,
      OffsetDateTime updatedAt) {}
}
